src/patch_parser.py

```diff
@@ -415,34 +433,41 @@ def _does_match_end_of_super_chunk(lines: InputSource) -> bool:
 def _parse_patch_file_unchanged_chunk(
         lines: InputSource,
         gerrit_orig_line: int,
-        gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
+        gerrit_new_line: int,
+        parser_state: HunkParserState) -> Tuple[int, int, PatchFileChunkLineMap]:
     in_start = lines.line_number()
     while (not _does_match_end_of_super_chunk(lines)) and (
             (not lines[0]) or (lines[0] and lines[0][0] != '+' and lines[0][0] != '-')):
         logging.info('dropping line: %s', lines[0])
         lines.consume()
         gerrit_orig_line += 1
         gerrit_new_line += 1
+        # logging.warning('Unchanged start: ' + str(in_start))
+        # logging.warning('Unchanged line: ' + lines[0])
+        # logging.warning('Unchanged lines - 1: ' + str(lines.line_number() - 1))
+    offset = gerrit_new_line - parser_state.get_number_of_deleted_lines() - lines.line_number()
     return (gerrit_orig_line,
             gerrit_new_line,
             PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
-                                  side='',
-                                  offset=(gerrit_new_line - lines.line_number())))
+                                  side='', offset=offset))
 
 
 def _parse_patch_file_added_chunk(
         lines: InputSource,
         gerrit_orig_line: int,
-        gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
+        gerrit_new_line: int,
+        parser_state: HunkParserState) -> Tuple[int, int, PatchFileChunkLineMap]:
     in_start = lines.line_number()
+    previous_number_of_removed_lines = parser_state.get_number_of_deleted_lines()
+    parser_state.reset_deleted_lines()
     logging.info('First char - 1: %c', lines[0][0])
     while lines[0] and lines[0][0] == '+':
-        previous = lines.get_previous_line()
         previous = lines.get_previous_line()
         lines.consume()
         gerrit_orig_line += 1
         gerrit_new_line += 1
-        if previous[0] == '-': # maybe add check to see if it is actually a modified one though I think this situation only applies when it actually is modified
+        if previous[
+            0] == '-':  # maybe add check to see if it is actually a modified one though I think this situation only applies when it actually is modified
```

Comment:
Nit: Can you make this a TODO comment and make it a wrapped line comment? This line is pretty long.